sdks/python/apache_beam/transforms/util_test.py

```diff
@@ -648,68 +648,70 @@ def test_find_empty(self):
-      result = (p | beam.Create(data)
-                | util.Regex.find_all("a(b*)", 1))
-      expected_retult = [['b', 'b', 'b'], ['bb', '', 'bbb']]
-      assert_that(result, equal_to(expected_retult))
+      test2 = (pcol | 'test 2' >> util.Regex.find_all("a(b*)", 1))
 
-    with TestPipeline() as p:
       # Find all with group 1 and remove empty items
-      result = (p | beam.Create(data)
-                | util.Regex.find_all("a(b*)", 1, outputEmpty=False))
-      expected_retult = [['b', 'b', 'b'], ['bb', 'bbb']]
-      assert_that(result, equal_to(expected_retult))
+      test3 = (pcol | 'test 3' >> util.Regex.find_all("a(b*)", 1,
+                                                      outputEmpty=False))
 
-    with TestPipeline() as p:
       # Find all with named group
-      result = (p | beam.Create(data)
-                | util.Regex.find_all("a(?P<namedgroup>b*)", 'namedgroup'))
-      expected_retult = [['b', 'b', 'b'], ['bb', '', 'bbb']]
-      assert_that(result, equal_to(expected_retult))
+      test4 = (pcol | 'test 4' >> util.Regex.find_all("a(?P<namedgroup>b*)",
+                                                      'namedgroup'))
 
-    with TestPipeline() as p:
       # Find all with named group with override Regex.ALL flag which return
       # all the groups in tuple format
-      result = (p | beam.Create(data)
-                | util.Regex.find_all("a(?P<namedgroup>b*)", util.Regex.ALL))
-      expected_retult = [[('ab', 'b'), ('ab', 'b'), ('ab', 'b')],
-                         [('abb', 'bb'), ('a', ''), ('abbb', 'bbb')]]
-      assert_that(result, equal_to(expected_retult))
+      test5 = (pcol | 'test 5' >> util.Regex.find_all("a(?P<namedgroup>b*)",
+                                                      util.Regex.ALL))
 
-    with TestPipeline() as p:
       # Find all with Regex.ALL flag and removes the empty groups items
-      result = (p | beam.Create(data)
-                | util.Regex.find_all("a(b*)", util.Regex.ALL,
-                                      outputEmpty=False))
-      expected_retult = [[('ab', 'b'), ('ab', 'b'), ('ab', 'b')],
-                         [('abb', 'bb'), ('abbb', 'bbb')]]
-      assert_that(result, equal_to(expected_retult))
+      test6 = (pcol | 'test 6' >> util.Regex.find_all("a(b*)", util.Regex.ALL,
+                                                      outputEmpty=False))
+
+      test1_expected = [['abb', 'a', 'abbb'], ['ab', 'ab', 'ab']]
+      test2_expected = [['b', 'b', 'b'], ['bb', '', 'bbb']]
+      test3_expected = [['b', 'b', 'b'], ['bb', 'bbb']]
+      test4_expected = [['b', 'b', 'b'], ['bb', '', 'bbb']]
+      test5_expected = [[('ab', 'b'), ('ab', 'b'), ('ab', 'b')],
+                        [('abb', 'bb'), ('a', ''), ('abbb', 'bbb')]]
+      test6_expected = [[('ab', 'b'), ('ab', 'b'), ('ab', 'b')],
+                        [('abb', 'bb'), ('abbb', 'bbb')]]
+
+      assert_that(test1, equal_to(test1_expected), label='assert test 1')
```

Comment:
I would still put the test expectations up next to the defines, and label them in a meaningful way. Probably getting rid of the intermediates would make this easier to read, e.g.

```
assert_that(
    pcol | 'group 1' >> util.Regex.find_all("a(b*)", group=1),
    equal_to([['b', 'b', 'b'], ['bb', '', 'bbb']]),
    label='CheckGroup1')

assert_that(
    pcol | 'group 1 non empty' >> util.Regex.find_all("a(b*)", group=1, outputEmpty=False),
    equal_to([['b', 'b', 'b'], ['bb','bbb']]),
    label='CheckGroup1NoEmpty')

...
```

Reply:
Sure, let me update them and put some meaningful labels.

As always, your feedback is very helpful. Thanks!